simple/rdm_tagged_peek.c

```diff
@@ -31,95 +31,187 @@
 #include <stdlib.h>
 #include <string.h>
 #include <getopt.h>
+#include <unistd.h>
 
 #include <rdma/fi_errno.h>
 #include <rdma/fi_tagged.h>
 
 #include <shared.h>
 
-static int tagged_peek(uint64_t tag)
+static int wait_for_send_comp(int count)
 {
+	int ret, completions = 0;
 	struct fi_cq_tagged_entry comp;
-	struct fi_msg_tagged msg;
+
+	do {
+		ret = fi_cq_sread(txcq, &comp, 1, NULL, -1);
+		if (ret != 1) {
+			FT_PRINTERR("fi_cq_sread", ret);
+			return ret;
+		}
+		completions++;
+	} while (completions < count);
+
+	return 0;
+}
+
+static int tag_queue_op(uint64_t tag, int recv, uint64_t flags)
+{
 	int ret;
+	struct fi_cq_tagged_entry comp;
+	struct fi_msg_tagged msg = {0};
+	struct fi_cq_err_entry cq_err;
```

Comment:
Please, add `memset(&cq_err, 0, sizeof(cq_err))` here to avoid unpredictable result of writing to invalid buffer (crash or something other).
It may happens due to statically allocated variable `cq_err` contains garbage in its fileds. One of this fields is recently added `err_data_size`. if this value will have some positive value, providers will consider that application provides some buffer for error reporting (I mean `err_data` field) to `fi_cq_readerr` and will utilize this buffer to take out provider's specific error-string or smth to this buffer.